tests/kernel/workq/work_queue_api/src/main.c

```diff
@@ -129,9 +130,15 @@ static void tdelayed_work_submit(void *data)
 			zassert_true(k_delayed_work_submit(&delayed_work[i],
 							   TIMEOUT) == 0, NULL);
 		}
-		/**TESTPOINT: check remaining timeout after submit*/
-		zassert_true(k_delayed_work_remaining_get(&delayed_work[i]) >=
-			     TIMEOUT, NULL);
+
+		time_remaining = k_delayed_work_remaining_get(&delayed_work[i]);
+
+		/**TESTPOINT: check remaining timeout after submit */
+		zassert_true(
+		    time_remaining <= __ticks_to_ms(z_ms_to_ticks(TIMEOUT)
+						    + _TICK_ALIGN) &&
+		    time_remaining >= __ticks_to_ms(z_ms_to_ticks(TIMEOUT) -
+						    z_ms_to_ticks(15)), NULL);
```

Comment:
regarding the ` - z_ms_to_ticks(15)` and the comments about time passed since submit
wouldn't it be more suitable to lock interrupts check time and apply the right time difference?
e.g.
```
irq_lock();
time_start = get_sys_time();
work_submit(&work, TO);
irq_unlock();

irq_lock();
time_meas = get_sys_time();
time_remain = work_remaining_get(&work);
irq_unlock();

assert( (time_remain <= (TO +tick)) && (time_remain >= TO - (time_meas - time_start + tick));
```
?

15 seems like a good thing now but will it be maintainable and work on all platforms this test run in the future?

Reply:
@andyross: Do you think that this proposal would be better?

For me, this suggest that you can freeze the time. In current implementation this is correct, but I do not know what will be result of work triggered by https://github.com/zephyrproject-rtos/zephyr/issues/17162.

Reply:
@andyross would you mind weighing in here?